File: detox/ios/DetoxTestRunner/Utilities/XCUIElement+Visibility.m
//
//  XCUIElement+Visibility.m
//  DetoxTestRunner
//
//  Created by Leo Natan (Wix) on 2/23/20.
//

#import "XCUIElement+Visibility.h"

@implementation XCUIElement (Visibility)

- (BOOL)dtx_isVisible
{
	return self.exists && self.isHittable && CGRectIsEmpty(self.frame) == NO;
}

@end

File: detox/ios/DetoxTestRunner/Utilities/XCUIElementExtensions/XCUIElement+Visibility.swift
//
//  XCUIElement+Visibility.swift
//  DetoxTestRunner
//
//  Created by Alon Haiut on 10/10/2021.
//  Copyright © 2021 Wix. All rights reserved.
//

import Foundation
import XCTest

extension XCUIElement {
	
	var dtx_isVisible: Bool {
		return self.exists && self.isHittable && self.frame.isEmpty == false;
	}
}
